### ISSUE TITLE
refactor(tauri): handle multiple redirects in update check

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#5383]
Found this in a customer's log. Did the version check succeed? Wondering if reqwest needs to be [configured to allow redirects](https://docs.rs/reqwest/latest/reqwest/redirect/struct.Policy.html).
```
{"time":"2024-06-14T02:11:53.355306781Z","target":"log","severity":"DEBUG","logFile":"/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/reqwest-0.12.4/src/async_impl/client.rs","logLine":2695,"logModulePath":"reqwest::async_impl::client","logTarget":"reqwest::async_impl::client","message":"redirect policy disallowed redirection to 'https://www.github.com/firezone/firezone/releases/download/gui-client-1.0.7/firezone-client-gui-linux_1.0.7_x86_64.deb'"}
{"time":"2024-06-14T02:11:53.355320758Z","target":"firezone_gui_client::client::updates","logging.googleapis.com/sourceLocation":{"file":"gui-client/src-tauri/src/client/updates.rs","line":"48"},"severity":"DEBUG","downloadUrl":"\"https://www.github.com/firezone/firezone/releases/download/gui-client-1.0.7/firezone-client-gui-linux_1.0.7_x86_64.deb\""}
{"time":"2024-06-14T02:11:53.355333282Z","target":"firezone_gui_client::client::updates","logging.googleapis.com/sourceLocation":{"file":"gui-client/src-tauri/src/client/updates.rs","line":"59"},"severity":"DEBUG","url":"Url { scheme: \"https\", cannot_be_a_base: false, username: \"\", password: None, host: Some(Domain(\"www.github.com\")), port: None, path: \"/firezone/firezone/releases/download/gui-client-1.0.7/firezone-client-gui-linux_1.0.7_x86_64.deb\", query: None, fragment: None }"}
{"time":"2024-06-14T02:11:53.35535869Z","target":"firezone_gui_client::client::gui","logging.googleapis.com/sourceLocation":{"file":"gui-client/src-tauri/src/client/gui.rs","line":"361"},"severity":"INFO","latestVersion":"Version { major: 1, minor: 0, patch: 7 }","message":"Our release is newer than, or the same as, the latest","ourVersion":"Version { major: 1, minor: 0, patch: 7 }"}
```

**Comment by ReactorScram** [comment]
BLUF: The code did work, but if we needed 2+ redirects it would misbehave.

It did succeed, it says on the last line that our version is 1.0.7 and the latest is 1.0.7 so no need to update.

It's a little odd but I do disable redirects. This code https://github.com/firezone/firezone/blob/eda1df56c2f7163ebee777981b60a8b070f0e7a0/rust/gui-client/src-tauri/src/client/updates.rs#L20-L47
does:
- Disable redirects
- Set up the user agent and URL
- Perform a `HEAD` request
- Expect a 307 temporary redirect 
- Parse the `location` header to see what the latest version is

It works as intended, but it would not work if we needed multiple redirects. To generalize it, I could use the default policy (10 redirects) and grab the final URL from `Reponse::url` instead. Not sure why I didn't do that. https://docs.rs/reqwest/latest/reqwest/struct.Response.html#method.url

**Comment by jamilbk** [comment]
Ah ok, no biggie then. It might be good to handle multiple just in case GH decides to start redirecting as well, but not a big priority at the moment.

**Comment by ReactorScram** [comment]
I looked into it and we can't do multiple redirects because it'll redirect to https://objects.githubusercontent.com and the version number will be hard to read.

The reason we only do 1 redirect is because it's just a fancy way to read `/website/redirects.js` off the server. We don't do the actual download from this URL, and if we needed to, we would turn redirects back on.

I'll open a PR to document that in a commit and clean it up, but I don't think the code itself needs to change
